File: src/whoami/paths.py
from __future__ import annotations

from pathlib import Path

from platformdirs import user_config_dir, user_data_dir


APP = "who-am-i"


def data_dir() -> Path:
    p = Path(user_data_dir(APP, "whoami"))
    p.mkdir(parents=True, exist_ok=True)
    return p


def config_dir() -> Path:
    p = Path(user_config_dir(APP, "whoami"))
    p.mkdir(parents=True, exist_ok=True)
    return p


def identity_path() -> Path:
    return config_dir() / "identity.yaml"


def settings_path() -> Path:
    return config_dir() / "settings.yaml"


def db_path() -> Path:
    return data_dir() / "whoami.db"


def reports_dir() -> Path:
    p = data_dir() / "reports"
    p.mkdir(parents=True, exist_ok=True)
    return p


def home() -> Path:
    return Path.home()

File: packages/core/src/paths.ts
import { homedir, platform } from "node:os";
import { join } from "node:path";
import { mkdirSync } from "node:fs";

const APP = "who-am-i";

export function home(): string {
  return homedir();
}

export function configDir(): string {
  const dir =
    platform() === "darwin"
      ? join(homedir(), "Library", "Application Support", APP)
      : platform() === "win32"
        ? join(process.env.APPDATA || join(homedir(), "AppData", "Roaming"), APP)
        : join(process.env.XDG_CONFIG_HOME || join(homedir(), ".config"), APP);
  mkdirSync(dir, { recursive: true });
  return dir;
}

export function dataDir(): string {
  if (platform() === "linux") {
    const dir = join(process.env.XDG_DATA_HOME || join(homedir(), ".local", "share"), APP);
    mkdirSync(dir, { recursive: true });
    return dir;
  }
  return configDir();
}

export function identityPath(): string {
  return join(configDir(), "identity.yaml");
}

export function settingsPath(): string {
  return join(configDir(), "settings.yaml");
}

export function dbPath(): string {
  return join(dataDir(), "whoami.db");
}

export function reportsDir(): string {
  const dir = join(dataDir(), "reports");
  mkdirSync(dir, { recursive: true });
  return dir;
}
